src/quo2/components/navigation/page_nav.cljs

```diff
@@ -0,0 +1,192 @@
+
+(def ^:private align-left
+  (assoc centrify-style :align-items :flex-start))
+
+(def ^:private icon-styles (assoc centrify-style :width 32 :height 32 :border-radius 10))
+
+(defn- big? [size] (= size :big))
+
+(defn- icon-props [color size]
+  (merge {:size      20
+          :container-style {:width  (if (big? size)
+                                      20
+                                      16)
+                            :height (if (big? size)
+                                      20
+                                      16)}}
+         (if-not (string/blank? color)
+           {:color    color}
+           {:no-color true})))
+
+(defn- mid-section-comp
+  [{:keys [mid-section-description-user-icon
+           horizontal-description?
+           text-secondary-color
+           align-mid?
+           text-color
+           mid-section-icon
+           mid-section-main-text
+           mid-section-type
+           mid-section-description]}]
+  [rn/view {:style (assoc
+                    centrify-style
+                    :flex-direction    :row
+                    :margin-horizontal 2)}
+   (when (or (and (not horizontal-description?)
+                  align-mid?
+                  (not= :text-with-description mid-section-type))
+             (and mid-section-description-user-icon
+                  (not mid-section-icon)))
+     [rn/image {:source {:uri mid-section-description-user-icon}
+                :style  {:width         32
+                         :height        32
+                         :border-radius 32
+                         :margin-right  8}}])
+   [rn/view {:style {:flex-direction (if horizontal-description?
+                                       :row
+                                       :column)}}
+    [text/text {:size   :paragraph-1
+                :weight :semi-bold
+                :style  {:color       text-color
+                         :line-height 21}}
+     mid-section-main-text]
+    (when mid-section-description
+      [text/text {:size   :paragraph-2
+                  :weight :medium
+                  :style  (cond-> {:padding-right 4
+                                   :color         text-secondary-color
+                                   :line-height   18}
+                            horizontal-description? (assoc :margin-left 4 :margin-top 2))}
+
+       mid-section-description])]])
+
+(defn- mid-section
+  [{:keys [horizontal-description? one-icon-align-left? mid-section-type left-align? mid-section-description mid-section-description-user-icon mid-section-main-text mid-section-right-icon mid-section-icon align-mid? mid-section-main-text-icon-color mid-section-left-icon]}]
+  (let [text-color                (if (colors/dark?) colors/neutral-5 colors/neutral-95)
+        text-secondary-color      (if (colors/dark?) colors/neutral-40 colors/neutral-50)
+        mid-section-comp-instance [mid-section-comp {:mid-section-description-user-icon mid-section-description-user-icon
```

Comment:
just wondering why you need to duplicate a map , why you can't use `mid-section` argument map ?

Reply:
Fixed